src/voodoo-prep/main.ml

```diff
@@ -43,12 +43,39 @@ module Prep = struct
       & opt (list (pair ~sep:':' string string)) []
       & info [ "u"; "universes" ] ~doc)
 
-  let cmd = Term.(const prep $ lib_dir $ switch $ universes)
+  let extra_universes =
+    let doc =
+      "Provide extra universes as 'package=universe id' couples. These will \
+       not be prepped but can be referenced by jsoo toplevels."
+    in
+    Arg.(
+      value
+      & opt (list (pair ~sep:':' string string)) []
+      & info [ "v"; "extra-universes" ] ~doc)
+
+  let cmd = Term.(const prep $ lib_dir $ switch $ universes $ extra_universes)
   let info = Term.info "prep" ~doc:"Prep a directory tree for compiling"
```

Comment:
@jonludlam I am sure there is a very good high-level explanation for why we now need `extra_universes` as a parameter to voodoo-prep.

Can you elaborate on that?